### PR TITLE
Add wallet connect support

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

**Review by nop33 — APPROVED** [review]
LGTM 👍  There are a few changes I would like to suggest, like using different array functions to find an element in an array and stuff like that, but I don't want to delay the process, code improvements can come later, let's make it work first!

File: packages/walletconnect/test/shared/WalletClient.ts
Comment (polarker) [inline packages/walletconnect/test/shared/WalletClient.ts:97]: Maybe we should restrict `networkId` to the three values (mainnet, testnet, devnet) only?

File: packages/web3-react/src/components/ConnectModal/ConnectWithWalletConnect.tsx
Comment (polarker) [inline packages/web3-react/src/components/ConnectModal/ConnectWithWalletConnect.tsx:41]: Let's set `_init = true` only when connect succeeds with `connect().then(...)`?

File: packages/web3-react/src/contexts/alephiumConnect.tsx
Comment (polarker) [inline packages/web3-react/src/contexts/alephiumConnect.tsx:41]: This can be removed now?

File: packages/web3-react/src/hooks/useAccount.tsx
Comment (polarker) [inline packages/web3-react/src/hooks/useAccount.tsx:28]: Just to confirm: this means we don't support auto-connect for WalletConnect right now?

File: packages/web3-react/src/types.ts
Comment (polarker) [inline packages/web3-react/src/types.ts:25]: Should we introduce a type for this id too?

**Review by polarker — COMMENTED** [review]
LGTM, just a few nits comments. Ready to merge once they are resoved.

File: packages/web3-react/src/components/Pages/Connectors/index.tsx
Comment (nop33) [reply packages/web3-react/src/components/Pages/Connectors/index.tsx:95]: > The previous name is Wallet Connect.

I couldn't find many references of "Wallet Connect" online. 

I can change the code back to its more readable form, no problem.

File: packages/web3-react/src/types.ts
Comment (nop33) [reply packages/web3-react/src/types.ts:25]: You mean an enum or sth similar? Sounds like a good idea 👍

File: packages/web3-react/src/components/Pages/Connectors/index.tsx
Comment (nop33) [reply packages/web3-react/src/components/Pages/Connectors/index.tsx]: updated!

Comment (polarker) [reply packages/web3-react/src/components/Pages/Connectors/index.tsx]: > I couldn't find many references of "Wallet Connect" online.

If it's on purpose change, then I am fine with it :)

File: packages/web3-react/src/hooks/useAccount.tsx
Comment (Lbqds) [reply packages/web3-react/src/hooks/useAccount.tsx:28]: Yes, because I'm not sure how we should support automatic connection via WalletConnect. I'll look into how to support this later.

File: packages/web3-react/src/types.ts
Comment (polarker) [reply packages/web3-react/src/types.ts:25]: Yes, but I am not sure whether it's doable due to injected connector

File: packages/web3-react/src/components/ConnectModal/ConnectWithInjector/index.tsx
Comment (nop33) [reply packages/web3-react/src/components/ConnectModal/ConnectWithInjector/index.tsx:95]: Ah, there is already one in the CLI! Nice!

Comment (nop33) [reply packages/web3-react/src/components/ConnectModal/ConnectWithInjector/index.tsx:95]: I would suggest that we export an array with the valid values so that we can do validation when needed, and also generate a type out of the array (so that we only define the valid list of values once). Here's how we can do that:

```ts
export const networkIds = ['mainnet', 'testnet', 'devnet'] as const
export type NetworkId = (typeof networkIds)[number]
```

We can update the `NetworkType` type from the CLI tool to this and use it everywhere else. WDYT?

Comment (Lbqds) [reply packages/web3-react/src/components/ConnectModal/ConnectWithInjector/index.tsx:95]: I agree with what you said, but I think we should add this to @alephium/web3. I'm working on that now.

Comment (polarker) [reply packages/web3-react/src/components/ConnectModal/ConnectWithInjector/index.tsx:95]: That's a great idea 👍 We use similar trick in WalletConnectProvider

File: packages/web3-react/src/hooks/useAccount.tsx
Comment (polarker) [reply packages/web3-react/src/hooks/useAccount.tsx:28]: Np, I will assign the task to @ross-weir later since he has been looking into this for quit some time and got blocked by us for a while.

**Pull request merged**
